network/dag/notifier.go

```diff
@@ -351,16 +383,20 @@ func (p *notifier) notifyNow(event Event) error {
 		}
 	}
 
-	if finished, err := p.receiver(*dbEvent); err != nil {
-		log.Logger().
-			WithError(err).
-			WithField(core.LogFieldTransactionRef, dbEvent.Hash.String()).
-			WithField(core.LogFieldEventSubscriber, p.name).
-			Errorf("Retry failed")
+	finished, err := p.receiver(*dbEvent)
+	if err != nil {
+		if errors.As(err, new(EventFatal)) {
```

Comment:
The previous if can be removed. This call also checks for nil

Reply:
This would mark an event with a non-fatal error and finished == true as completed. This changes the current logic of the method